src/duration/__tests__/format.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019 JS Foundation and other contributors
+// Copyright 2024 YANDEX LLC
+
+import {duration} from '..';
+import {settings} from '../../settings';
+
+const dur = () =>
+    duration({
+        years: 1,
+        months: 2,
+        weeks: 1,
+        days: 3,
+        hours: 4,
+        minutes: 5,
+        seconds: 6,
+        milliseconds: 7,
+    });
+
+//------
+// #toISOString()
+//------
+
+test('Duration#toISOString fills out every field', () => {
+    expect(dur().toISOString()).toBe('P1Y2M1W3DT4H5M6.007S');
+});
+
+test('Duration#toISOString fills out every field with fractional', () => {
+    const dur = duration({
+        years: 1.1,
+        months: 2.2,
+        weeks: 1.1,
+        days: 3.3,
+        hours: 4.4,
+        minutes: 5.5,
+        seconds: 6.6,
+        milliseconds: 7,
+    });
+    expect(dur.toISOString()).toBe('P1.1Y2.2M1.1W3.3DT4.4H5.5M6.607S');
+});
+
+test('Duration#toISOString creates a minimal string', () => {
+    expect(duration({years: 3, seconds: 45}).toISOString()).toBe('P3YT45S');
+    expect(duration({months: 4, seconds: 45}).toISOString()).toBe('P4MT45S');
+    expect(duration({months: 5}).toISOString()).toBe('P5M');
+    expect(duration({minutes: 5}).toISOString()).toBe('PT5M');
+});
+
+test('Duration#toISOString handles negative durations', () => {
+    expect(duration({years: -3, seconds: -45}).toISOString()).toBe('P-3YT-45S');
```

Comment:
What does negative value mean?

Reply:
Negative durations are described in ISO 8601-2:

```
[ 11.3.2 Composite representation ]
| The composite representation of a duration is a more flexible and
| relaxed specification for duration than that of .ISO 8601-1:2019,
| 5.5.2. It accepts all expressions of the duration representation given
| in ISO 8601-1:2019, 5.5.2 and is given as follows.
|
| [!]["-"]["P"][ durationUnits(m)]
|
| where [durationUnits(m)] contains time scale components for expressing
| (positive or negative) duration (see 11.2).
|
| Expressions in the two examples below are valid in ISO 8601-1.
|
| EXAMPLE 1 'P3D', duration of three days.
| EXAMPLE 2 'P180Y800D', duration of one-hundred-and-eighty years and eight-hundred days.
|
| Expressions in the following four examples below are not valid in ISO
| 8601-1, but are valid as specified in this clause.
|
| EXAMPLE 3 'P3W2D', duration of three weeks and two days, which is 23 days (equivalent to the expression
| 'P23D'). In ISO 8601-1, ["W"] is not permitted to occur along with any other component.
| EXAMPLE 4 'PSYlOW', duration of five years and ten weeks.
| EXAMPLE 5 'P-3M-3DT1HSM', duration of three months and three days in the reverse direction, with one hour
| and five minutes in the original direction.
| EXAMPLE 6 'P-ZM-1D', duration in the reverse direction of two months and one day.
|
| When a minus sign is provided as prefix to the duration designator
| ["P"], the minus sign can be internalized into individual time scale
| components within the duration expression by applying to every time
| scale component within.
|
| EXAMPLE 7 '-P2M1D' is equivalent to 'P-2M-1D'.
| EXAMPLE 8 '-P5DT10H' is equivalent to 'P-5DT-10H'.
|
| When a minus sign is applied to a time scale component whose value is
| already negative (pointing to the reverse direction), it means that
| the direction of duration should be once again reversed and should be
| turned into a positive value.
|
| EXAMPLE 9 '-P8M-1D', duration in reverse, "eight months minus one day", is equivalent to 'P-8M1D', "eight
| months ago with a day ahead".
| EXAMPLE 10 '-P-5WT-18H30M', duration in reverse, "go back five weeks, eighteen hours but thirty minutes
| ahead", is equivalent to 'P5WT18H-30M', "go ahead five weeks, eighteen hours, but thirty minutes back".
```